libfwupdplugin/fu-device.c

```diff
@@ -3228,9 +3240,10 @@ fu_device_finalize (GObject *object)
 		g_object_unref (priv->quirks);
 	if (priv->poll_id != 0)
 		g_source_remove (priv->poll_id);
+	if (priv->metadata != NULL)
```

Comment:
this looks like you actually caught a bug with this change too!  It shouldn't have cleared it when the mutex was released.

Reply:
100% unintentional :)

Reply:
Since it's the case for both of the mutex use, maybe double check that it wasn't already right. `g_rw_lock_clear` releases the mutex, but if it wasn't released will trying to clear cause a deadlock?  or did this function take the mutex at the start?

Reply:
> but if it wasn't released will trying to clear cause a deadlock

It's apparently _undefined_. I'll do some review.